js/tidelinedata.js

```diff
@@ -526,6 +526,37 @@ function TidelineData(data, opts) {
   });
   endTimer('diabetesData');
 
+  startTimer('deviceEvents');
+  var parameters = _.filter( data,  {type: 'deviceEvent', subType: 'deviceParameter'});
+  var sortedParameters =_.orderBy(parameters,['normaltime'], ['desc']);
+
+  this.deviceParameters = [];
+  if (sortedParameters.length > 1) {
+    var first = sortedParameters[0];
+    var group = { 
+      normalTime: first.normalTime,
+      id: first.id,
+      params: [first]
+    }
+    for (var i = 1; i < sortedParameters.length; ++i) {      
```

Comment:
```suggestion
    for (let i = 1; i < sortedParameters.length; ++i) {      
```
Avoid using `var` in this big function, or change to a more explicit (and unique) name.
In this configuration a bug will be difficult to spot.